Source/WebCore/Modules/compression/CompressionStreamEncoder.cpp

```diff
@@ -164,4 +176,82 @@ ExceptionOr<Ref<JSC::ArrayBuffer>> CompressionStreamEncoder::compress(std::span<
 
     return compressedData.releaseNonNull();
 }
+
+
+#if PLATFORM(COCOA)
+ExceptionOr<bool> CompressionStreamEncoder::initializeAppleCompressionFramework() 
+{
+    m_initialized = true;
```

Comment:
I think its possible to both fail to initialize the framework and still try to use it later by putting this here. I think this `m_initialized` variable shouldn't be set until after we've checked the return value from `compression_stream_init`.